src/D2L.Bmx/BmxConfigProvider.cs

```diff
@@ -15,36 +12,42 @@ internal class BmxConfigProvider( FileIniDataParser parser ) : IBmxConfigProvide
 	public BmxConfig GetConfiguration() {
 		// Main config is at ~/.bmx/config
 		string configFileName = BmxPaths.CONFIG_FILE_NAME;
-
+		var data = new IniData();
+		if( File.Exists( configFileName ) ) {
+			try {
+				var tempdata = parser.ReadFile( configFileName );
+				data.Merge( tempdata );
+			} catch( Exception ) {
+				Console.Write( "Error occur while parsing config file." );
+			}
+		}
 		// If set, we recursively look up from CWD (all the way to root) for additional bmx config files (labelled as .bmx)
-		var configBuilder = new ConfigurationBuilder().AddIniFile( configFileName, optional: true );
-
 		FileInfo? projectConfigInfo = GetProjectConfigFileInfo();
 
 		if( projectConfigInfo is not null ) {
-			// Default file provider ignores files prefixed with ".", we need to provide our own as a result
-			var fileProvider =
-				new PhysicalFileProvider( projectConfigInfo.DirectoryName!, ExclusionFilters.None );
-			configBuilder.AddIniFile( fileProvider, projectConfigInfo.Name, optional: false,
-				reloadOnChange: false );
+			string configFilePath = Path.Combine( projectConfigInfo.DirectoryName ?? "", projectConfigInfo.Name );
+			try {
+				var tempdata = parser.ReadFile( configFilePath );
```

Comment:
we don't need to reconstruct the file path
```suggestion
			try {
				var tempdata = parser.ReadFile( projectConfigInfo.FullName );
```